subprojects/model-core/src/main/java/org/gradle/api/internal/provider/MappingProvider.java

```diff
@@ -67,8 +68,17 @@ public ExecutionTimeValue<? extends OUT> calculateExecutionTimeValue() {
         } else if (value.isMissing()) {
             return ExecutionTimeValue.missing();
         } else {
-            return ExecutionTimeValue.fixedValue(transformer.transform(value.getFixedValue()));
+            // not calling ExecutionTimeValue.fixedValue() directly, because the value may have a side effect
+            return ExecutionTimeValue.value(checkTransformedValue(value.toValue().transform(transformer)));
+        }
+    }
+
+    private Value<OUT> checkTransformedValue(Value<OUT> transformedValue) {
```

Comment:
Why do we need this method now?

Reply:
See here: https://github.com/gradle/gradle/pull/21491#discussion_r943261874